config.py

```diff
@@ -34,57 +34,92 @@
 c['slavePortnum'] = 9989
 
 from flocker_bb.password import generate_password
-from flocker_bb.ec2_buildslave import EC2BuildSlave
+from flocker_bb.ec2 import rackspace_slave, ec2_slave
 from buildbot.buildslave import BuildSlave
 
-cloudInit = FilePath(__file__).sibling("slave").child("cloud-init.sh").getContent()
+cloudInit = FilePath(__file__).sibling("slave").child(
+    "cloud-init.sh").getContent()
 
 c['slaves'] = []
 SLAVENAMES = {}
 
+
+def get_cloud_init(name, base, password, provider, privateData, slavePortnum):
+    """
+    :param bytes name: The name of the buildslave.
+    :param bytes base: The name of image used for the buildslave.
+    :param bytes password: The password that the buildslave will use to
+       authenticate with the buildmaster.
+    :param bytes provider: The cloud ``provider`` hosting the buildslave. This
+        is added to an environment variable, so that cloud ``provider``
+        specific tests know which cloud authentication plugin to load and which
+        credentials to load from the credentials file.
+    :param dict privateData: The non-slave specific keys and values from the
+        buildbot ``config.yml`` file.
+    :parma int slavePortnum: The TCP port number on the buildmaster to which
+        the buildslave will connect.
+    :returns: The ``bytes`` of a ``cloud-init.sh`` script which can be supplied
+       as ``userdata`` when creating an on-demand buildslave.
+    """
```

Comment:
This seems to work okay as-is for now. Happy to see it converted to something better later but ... later.